### DISCUSSION TITLE
Query string gets stripped out when using the logged_in scheme.

### DISCUSSION THREAD

**Discussion by sunscreem** [statamic/cms#9530] [Q&A] [answered]
Hi All,

When my users visit

https://mysite.com/someurl?token=1234

that page is protected using the usual `logged_in` scheme.

They therefore get redirect to:

https://mysite.com/loginpage?redirect=https://mysite.com/someurl

Note that the `?token=1234` has been stripped out.

Ideally I want them to be redirected to 

https://mysite.com/loginpage?redirect=https://mysite.com/someurl?token=1234

Any ideas why this might be happening?

**Comment by jasonvarga** [comment, marked as answer]
Nice find. Yes, it's happening because we use `request()->url()` here where we should probably be using `fullUrl()`

https://github.com/statamic/cms/blob/d76a4b4d0743e809b0f63485e9d40be980787f4c/src/Auth/Protect/Protection.php#L68-L71

Can you make an issue instead?
